### PR TITLE
Leo's Loo Too / MEET: fix weight scale, decode notifications, add MEET variant

### PR DESCRIPTION
**Pull request by algrym**
## Leo's Loo Too / MEET self-cleaning cat toilet — corrected config + notification decode

This corrects and extends the `leos_loo_too.yaml` config (Tuya category `msp`)
against live data from **four physical units** — three "Leo's Loo Too"
(`1badzyvwh1e1hrog`) and one rebadged "MEET" (`kw4ztiq3kxmcf8dx`) on identical
hardware — plus the Tuya cloud values as a reference oracle, and a multi-day
passive capture of the device's async report stream that decoded the DP21
notification bitfield.

### Changes

| Change | Confidence | Evidence |
| --- | --- | --- |
| **Weight DP6 scale `0.1` → `10`** | high | Raw is tenths of a kg (42 → 4.2 kg, matches cloud). The old `0.1` produced 420 kg. Consistent with the other weight litterbox configs (`duoqu`, `vevor`, `homall` all use `scale: 10`). |
| **Add the `kw4ztiq3kxmcf8dx` product id** | high | One unit enrols under this second product id. Owner-confirmed it is the *same* physical Leo's Loo Too hardware carrying the MEET / Chinese-market firmware profile — not a separate MEET product — so it stays under the `Leo's Loo` brand with model `Too (MEET firmware)`. |
| **DP2 (mode) / DP5 (delay) → `optional`** | high | Absent on that firmware variant (reports neither). |
| **DP22 fault → problem binary_sensor** | high | Same DP/pattern as the sibling `sailesi_litterbox` config. Also carries the raw DP21 notification code as a `notification_code` attribute (per review — see below). |
| **Decode DP21 notification bitfield** | high | See bit table below — confirmed by correlating the async report stream against observed device behaviour across all four units over several days. |
| **Keep DP7 "times used today", hidden** | high | Dead constant (`=1`) on every observed unit; never updates locally (also `1` in #1840's dump), so the real per-use count is not exposed on the LAN. Per review, rather than removing it (which would orphan the entity in existing installs) it is kept with an `available` null-test + `hidden: unavailable`, so it hides only where the DP is absent. |
| **Keep DP16 "light", hidden** | high | Never reported on the local protocol, so it only stranded an `unavailable` entity — but per review it is kept with `hidden: true` rather than removed, to avoid orphaning it in existing registries. |

### DP21 notification bits (decoded)

| Bit (value) | Meaning | Exposed as |
| --- | --- | --- |
| bit0 (1) | **Bin full / needs emptying** — box goes amber and *latches* this bit until the drawer is emptied and the reset button pressed | `binary_sensor` "Bin full" (plain on/off state) |
| bit1 (2) | Manual / commanded clean rotation active | folded into "Cleaning" (class `running`, mask 6) |
| bit2 (4) | Auto-clean rotation active | folded into "Cleaning" (class `running`, mask 6) |
| bit6 (64) | UV sterilisation phase active | `binary_sensor` "UV sterilizing" (class `running`) |

The raw notification code is retained as a `notification_code` attribute on the
DP22 Fault sensor (per review — not its own entity) so any remaining undecoded
bits stay observable. These are exposed as plain device **states** —
"bin full" is routine operation, not a malfunction, so no `problem`/alert
semantics are baked in; alerting is left to the user's own automations. (The
separate DP22 **Fault** binary_sensor keeps `class: problem`, as a fault is a
genuine device malfunction.)

The **bit0 "bin full"** decode was ground-truthed: an owner found the physical
amber "needs emptying" ring on a box while DP21 read exactly `1`; pressing the
reset button cleared it (`1 → 2 → …`), all signalled through DP21.

The **bit1 vs bit2** distinction was ground-truthed with a controlled test: a
timed manual-clean button press drove `DP21 0 → 2`, and **`2` held for 71 s** —
the full clean rotation, watched on the device — before clearing to `0`, with
the UV phase (`64`) following 6 s later. That is the identical lifecycle to a
bit2 auto-clean, just signalled on bit1, so **both bits mean "a clean rotation
is active"** and "Cleaning" masks them together (mask 6). Note: the device's own
Tuya data model labels bit1 `excessive_clean_times`, but the sustained 71-second
hold shows it is a clean-active state on this firmware, not a momentary warning —
so this decode follows the observed behaviour, not that label. (The DP22 fault
bit labels `motor`/`program`/`g_sensor` from the same model are left unverified,
as no fault fired on any unit; only the raw fault code is exposed.)

### Notes for the maintainer
- New binary_sensors use `name:` rather than `translation_key:` — happy to
  switch to translation keys (with `strings.json` additions) if preferred.
- Prior art: #1840 has another owner's raw dump corroborating DP7=1 and DP101=15.

### CODE REVIEW

File: custom_components/tuya_local/devices/leos_loo_too.yaml [hunk 1]
```diff
@@ -69,12 +109,84 @@ entities:
...
+        type: bitfield
+        name: sensor
+        optional: true
+        mapping:
+          - dps_val: 64
+            value: true
+          - value: false
+  # Raw notification code retained for any bits not yet decoded.
+  - entity: sensor
+    name: Notification code
```
Comment (make-all) [inline custom_components/tuya_local/devices/leos_loo_too.yaml]: This is not really useful, so can just be an attribute on another entity (I suggest the problem binary sensor, as attaching it to one of the notification flags will be less obvious if you ever want to fetch it.)

File: custom_components/tuya_local/devices/leos_loo_too.yaml [hunk 2]
```diff
@@ -1,16 +1,51 @@
...
+name: Leo's Loo Too
 products:
   - id: 1badzyvwh1e1hrog
     manufacturer: "Leo's Loo"
     model: Too
+  - id: kw4ztiq3kxmcf8dx
+    manufacturer: "Leo's Loo"
+    model: Too (MEET)
 entities:
-  - entity: sensor
```
Comment (make-all) [inline custom_components/tuya_local/devices/leos_loo_too.yaml:7]: Removing an entity from a released config will leave a zombie entity in the registry.

It may be better to keep it and add `hidden: true` instead.

File: custom_components/tuya_local/devices/leos_loo_too.yaml [hunk 3]
```diff
@@ -1,16 +1,51 @@
...
+#            but is never reported on the local protocol, so it only ever
+#            stranded a permanently-unavailable entity.
+name: Leo's Loo Too
 products:
   - id: 1badzyvwh1e1hrog
     manufacturer: "Leo's Loo"
     model: Too
+  - id: kw4ztiq3kxmcf8dx
+    manufacturer: "Leo's Loo"
+    model: Too (MEET)
```
Comment (make-all) [inline custom_components/tuya_local/devices/leos_loo_too.yaml]: The brand for this one is MEET right? Not Leo's Loo.

File: custom_components/tuya_local/devices/leos_loo_too.yaml [hunk 4]
```diff
@@ -40,13 +77,14 @@ entities:
       - id: 13
         name: switch
         type: boolean
-  - entity: light
+  - entity: switch
+    translation_key: sleep
     category: config
+    class: switch
     dps:
-      - id: 16
```
Comment (make-all) [inline custom_components/tuya_local/devices/leos_loo_too.yaml:46]: The original author put this in there, so their variant may be supporting this. Rather than removing it, add an available attribute that tests if the dps_val is null, and add `hidden: unavailable` to the entity

**Review by make-all — CHANGES_REQUESTED** [review]

File: custom_components/tuya_local/devices/leos_loo_too.yaml (hunk 1 above)
Comment (algrym) [reply custom_components/tuya_local/devices/leos_loo_too.yaml]: Dropped the standalone sensor and moved the raw DP21 code onto the Fault (problem) binary_sensor as a `notification_code` attribute, so it stays available for decoding any leftover bits without adding an entity.

File: custom_components/tuya_local/devices/leos_loo_too.yaml (hunk 2 above)
Comment (algrym) [reply custom_components/tuya_local/devices/leos_loo_too.yaml:7]: Kept the DP16 `light` entity and added `hidden: true` rather than removing it. Agreed on not orphaning entities in existing installs.

File: custom_components/tuya_local/devices/leos_loo_too.yaml (hunk 3 above)
Comment (algrym) [reply custom_components/tuya_local/devices/leos_loo_too.yaml]: Some Leo's Loo Too units come with the MEET (Chinese-market) firmware, which enrols under the separate `kw4ztiq3kxmcf8dx` product id. Same physical hardware, not a distinct MEET product. I've kept it under manufacturer "Leo's Loo" with model "Too (MEET firmware)" and added a comment explaining it. Happy to relabel if you'd rather it track the product-id branding instead.

File: custom_components/tuya_local/devices/leos_loo_too.yaml (hunk 4 above)
Comment (algrym) [reply custom_components/tuya_local/devices/leos_loo_too.yaml:46]: Kept DP7 with an `available` dp that tests the raw value for null plus `hidden: unavailable`, so it stays visible on variants that report it and hides itself where the dp is absent. On my units it reads a constant `1`, but this keeps the door open for variants where it carries a real count.

File: custom_components/tuya_local/devices/leos_loo_too.yaml [hunk 5]
```diff
@@ -1,16 +1,68 @@
-name: Pet toilet
+# Leo's Loo Too / MEET self-cleaning cat toilet  (Tuya category "msp")
```
Comment (make-all) [inline custom_components/tuya_local/devices/leos_loo_too.yaml]: The added comments are excessive. You should not be explaining what you have done in file comments, that is what commit comments are for.

File: custom_components/tuya_local/devices/leos_loo_too.yaml [hunk 6]
```diff
@@ -1,16 +1,68 @@
...
+#   high   - DP7 "times used today" (excretion_times_day) reads a constant 1 on
+#            the local protocol: a per-use pulse the cloud sums server-side, so
+#            the local value never changes. Rather than remove it (orphaning the
+#            entity in existing installs) it is kept, and hidden when the dp is
+#            absent (available null-test + hidden: unavailable). #1840's dump
+#            shows =1 too.
+#   medium - DP16 "light" is never reported on the local protocol here. It
+#            shipped in the released config, so it is kept but marked
+#            hidden: true rather than removed (removal would orphan the entity).
+name: Leo's Loo Too
```
Comment (make-all) [inline custom_components/tuya_local/devices/leos_loo_too.yaml]: This change is incorrect. It will affect the name shown for other devices that happen to match the same config

File: custom_components/tuya_local/devices/leos_loo_too.yaml [hunk 7]
```diff
@@ -69,12 +137,106 @@ entities:
...
+        class: measurement
+        mapping:
+          # Raw value is in tenths of a kg (42 -> 4.2 kg, cloud-confirmed).
+          - scale: 10
+  # DP7 "times used today" (excretion_times_day) reads a constant 1 on the local
+  # protocol here - a per-use pulse the cloud sums server-side, so the local
+  # value never changes. It shipped in the released config and another variant
+  # may report it, so rather than remove it (which would orphan the entity in
+  # existing installs) it is kept, and hidden when the dp is absent.
+  - entity: sensor
```
Comment (make-all) [inline custom_components/tuya_local/devices/leos_loo_too.yaml]: Please move this back to where it was to minimize changes and make it clearer for review what you actually changed here.

**Review by make-all — CHANGES_REQUESTED** [review]

File: custom_components/tuya_local/devices/leos_loo_too.yaml (hunk 5 above)
Comment (algrym) [reply custom_components/tuya_local/devices/leos_loo_too.yaml]: Done — dropped the header changelog and trimmed the inline comments to bare device facts; the rationale lives in the PR description and commit history now.

File: custom_components/tuya_local/devices/leos_loo_too.yaml (hunk 6 above)
Comment (algrym) [reply custom_components/tuya_local/devices/leos_loo_too.yaml]: Reverted — the top-level name is back to "Pet toilet".

File: custom_components/tuya_local/devices/leos_loo_too.yaml (hunk 7 above)
Comment (algrym) [reply custom_components/tuya_local/devices/leos_loo_too.yaml]: Moved "Times used today" back to its original position; the diff there is now just the added hidden/available DPs in place.
